src/langchain_google_cloud_sql_mysql/mysql_loader.py

```diff
@@ -32,48 +32,75 @@ def _parse_doc_from_table(
-        doc = Document(page_content=page_content, metadata=metadata)
+        row_data = {column: getattr(row, column) for column in column_names}
+        doc = _parse_doc_from_row(content_columns, metadata_columns, row_data)
         docs.append(doc)
     return docs
 
 
+def _parse_doc_from_row(
+    content_columns: Iterable[str], metadata_columns: Iterable[str], row: Dict
+) -> Document:
+    page_content = " ".join(
+        str(row[column]) for column in content_columns if column in row
+    )
+    metadata: Dict[str, Any] = {}
+    # load metadata from langchain_metadata column
+    if row.get(DEFAULT_METADATA_COL):
+        extra_metadata = json.loads(row[DEFAULT_METADATA_COL])
+        for k, v in extra_metadata.items():
+            if DEFAULT_METADATA_COL in metadata_columns or k in metadata_columns:
+                metadata[k] = v
+    # load metadata from other columns
+    for column in metadata_columns:
+        if column in row and column != DEFAULT_METADATA_COL:
+            metadata[column] = row[column]
+    return Document(page_content=page_content, metadata=metadata)
+
+
+def _parse_row_from_doc(column_names: Iterable[str], doc: Document) -> Dict:
+    doc_metadata = doc.metadata.copy()
+    row: Dict[str, Any] = {"page_content": doc.page_content}
+    for entry in doc.metadata:
+        if entry in column_names:
+            row[entry] = doc_metadata[entry]
+            del doc_metadata[entry]
+    # store extra metadata in langchain_metadata column in json format
+    if DEFAULT_METADATA_COL in column_names and len(doc_metadata) > 0:
+        row[DEFAULT_METADATA_COL] = doc_metadata
+    return row
+
+
 class MySQLLoader(BaseLoader):
     """A class for loading langchain documents from a Cloud SQL MySQL database."""
 
     def __init__(
         self,
         engine: MySQLEngine,
-        query: str,
+        table_name: str = "",
+        query: str = "",
         content_columns: Optional[List[str]] = None,
         metadata_columns: Optional[List[str]] = None,
     ):
         """
         Args:
           engine (MySQLEngine): MySQLEngine object to connect to the MySQL database.
-          query (str): The query to execute in MySQL format.
+          table_name (str): The MySQL database table name. (OneOf: table_name, query).
+          query (str): The query to execute in MySQL format.  (OneOf: table_name, query).
           content_columns (List[str]): The columns to write into the `page_content`
              of the document. Optional.
           metadata_columns (List[str]): The columns to write into the `metadata` of the document.
              Optional.
         """
         self.engine = engine
+        self.table_name = table_name
         self.query = query
         self.content_columns = content_columns
         self.metadata_columns = metadata_columns
+        if self.table_name == "" and self.query == "":
+            raise ValueError("One of table_name or query needs be specified.")
```

Comment:
empty strings are falsy values in Python so lets use the `not` here for a more Pythonic approach
```suggestion
        if not self.table_name and not self.query:
            raise ValueError("One of 'table_name' or 'query' must be specified.")
```

Reply:
Thanks for suggestion. Updated.